src/pages/StudyScreen/index.jsx

```diff
@@ -119,16 +129,47 @@ const StudyScreen = ({ route }) => {
       accessToken.current = token;
       setConfig(finalConfig);
 
-      const wMap = route.params.type === WORD_LIST_TYPE.COLLECTED ? filterCollectedWords(allWordsMap, finalConfig.collectedWords) : allWordsMap;
-      const wMapKeys = shuffleArray(Object.keys(wMap));
-      setWordsMap(wMap);
-      setWordsMapKeys(wMapKeys);
-      setWordData(wMap[wMapKeys[currIdx]]);
+      const wList = route.params.type === WORD_LIST_TYPE.COLLECTED ? extractCollectedWordsByTime(allWordsObject, finalConfig.collectedWords) : allWordsList;
+      setWordList(wList);
+      setWordData(wList[wordIndex]);
       setLoading(false);
     };
     setup();
   }, []);
 
+  useUpdateEffect(() => {
+    let newWordList = [];
+    if (shuffle) {
+      newWordList = shuffleArray(wordList);
+    } else if (route.params.type === WORD_LIST_TYPE.COLLECTED) {
+      newWordList = extractCollectedWordsByTime(allWordsObject, config.collectedWords);
+    } else {
+      newWordList = [...wordList].sort((w1, w2) => {
+        if (w1.word > w2.word) return 1;
+        if (w1.word < w2.word) return -1;
+        return 0;
+      });
+    }
+    setWordList(newWordList);
+    setWordIndex(0);
+    setWordData(newWordList[0]);
+  }, [shuffle]);
+
+  useUpdateEffect(() => {
+    if (route.params.type === WORD_LIST_TYPE.COLLECTED || shuffle) {
+      return;
+    }
+    const newWordList = allWordsList.sort((w1, w2) => {
```

Comment:
Create a function of sorting word list by alphabetical order since this sorting appears multiple times.